projects/gloo/api/v1/enterprise/options/ai/ai.proto

```diff
@@ -9,714 +9,738 @@ option (extproto.equal_all) = true;
     string deployment_name = 4;
   }
 
+  // Configuration for the backend LLM provider authentication token.
   oneof embedding {
-    // OpenAI embedding
+    // Embedding settings for the OpenAI provider.
     OpenAI openai = 1;
-    // Azure OpenAI embedding
+    // Embedding settings for the Azure OpenAI provider.
     AzureOpenAI azure_openai = 2;
   }
 }
 
-// Settings for the Semantic Caching feature
+// Cache previous model responses to provide faster responses to similar requests in the future.
+// Results might vary depending on the embedding mechanism used, as well
+// as the similarity threshold set. Semantic caching reduces the number of requests
+// to the LLM provider, improves the response time, and reduces costs.
+//
+// Example configuring a route to use a `redis` datastore and OpenAI for RAG:
+// ```yaml
+// semanticCache:
+//   datastore:
+//     redis:
+//       connectionString: redis://172.17.0.1:6379
+//   embedding:
+//     openai:
+//       authToken:
+//         secretRef:
+//           name: openai-secret
+//           namespace: gloo-system
+// ```
 message SemanticCache {
 
-  // Settings for the Redis database
+  // Settings for a Redis database.
   message Redis {
-    // Connection string to the Redis database
+    // Connection string to the Redis database, such as `redis://172.17.0.1:6379`.
     string connection_string = 1;
 
     // Similarity score threshold value between 0.0 and 1.0 that determines how similar
-    // two queries need to be in order to return a cached result.
-    // The lower the number, the more similar the queries need to be for a cache hit.
+    // two queries must be in order to return a cached result.
+    // The lower the number, the more similar the queries must be for a cache hit.
     //
     // +kubebuilder:validation:Minimum=0
     // +kubebuilder:validation:Maximum=1
     float score_threshold = 2;
   }
 
-  // Settings for the Weaviate database
+  // Settings for a Weaviate database.
   message Weaviate {
-    // Connection string to the Weaviate database, scheme should NOT be included.
-    // For example: weaviate.my-ns.svc.cluster.local
-    // NOT: http://weaviate.my-ns.svc.cluster.local
+    // Connection string to the Weaviate database.
+    // Do not include the scheme. For example, the format
+    // `weaviate.my-ns.svc.cluster.local` is correct. The format
+    // `http://weaviate.my-ns.svc.cluster.local`, which includes the scheme, is incorrect.
     string host = 1;
-    // HTTP port to use, if unset will default to 8080
+    // HTTP port to use. If unset, defaults to `8080`.
     uint32 http_port = 2;
-    // GRPC port to use, if unset will default to 50051
+    // GRPC port to use. If unset, defaults to `50051`.
     uint32 grpc_port = 3;
-    // Whether or not to use a secure connection, true by default
+    // Whether to use a secure connection. Defaults to `true`.
```

Comment:
This is a bit confusing and a bit against the convention where the default is always 0, "" or false. In the code, I also do not find where we set this to true by default (maybe I missed it). Maybe the original comment means secure connection by default? @EItanya can you confirm?

Also, it looks like this setting applies to both the http and grpc connections.

Reply:
Correct, it is the comment here that's confusing. The connection is secure by default